src/__tests__/command.spec.ts

```diff
@@ -219,24 +219,51 @@ describe('buildkite-graph', () => {
                 ),
             ]);
 
-            createTest('plugins', () =>
-                new Pipeline('whatever').add(
-                    new CommandStep('noop').plugins
-                        .add(
-                            new Plugin('bugcrowd/test-summary#v1.5.0', {
-                                inputs: [
-                                    {
-                                        label: ':htmllint: HTML lint',
-                                        artifact_path:
-                                            'web/target/htmllint-*.txt',
-                                        type: 'oneline',
-                                    },
-                                ],
-                            }),
-                        )
-                        .plugins.add(new Plugin('detect-clowns#v1.0.0')),
-                ),
-            );
+            describe('plugins', () => {
+                const plugins = [
+                    new Plugin('bugcrowd/test-summary#v1.5.0', {
+                        inputs: [
+                            {
+                                label: ':htmllint: HTML lint',
+                                artifact_path: 'web/target/htmllint-*.txt',
+                                type: 'oneline',
+                            },
+                        ],
+                    }),
+                    new Plugin('detect-clowns#v1.0.0'),
+                ];
+                const stepWithPlugins = new CommandStep('noop').plugins
+                    .add(plugins[0])
+                    .plugins.add(plugins[1]);
+
+                createTest('add plugins', () =>
+                    new Pipeline('whatever').add(stepWithPlugins),
+                );
+
+                it.each([
+                    {
+                        predicate: (plugin: Plugin) =>
+                            plugin.pluginNameOrPath.includes('bugcrowd'),
+                        expected: [plugins[0]],
+                    },
+                    {
+                        predicate: (plugin: Plugin) =>
+                            plugin.pluginNameOrPath.includes('.0'),
+                        expected: plugins,
+                    },
+                    {
+                        predicate: (plugin: Plugin) =>
+                            plugin.pluginNameOrPath.includes(
+                                'NONEXISTENT_NAME',
+                            ),
+                        expected: [],
+                    },
+                ])('filter plugins', ({ predicate, expected }) => {
+                    expect(
+                        stepWithPlugins.plugins.filter(predicate),
+                    ).toMatchObject(expected);
+                });
```

Comment:
These tests are mostly testing the `Array.filter` implementation of the JS runtime :) I think a single test to ensure that the API works would be sufficient.
Something like:

```suggestion
                it('is possible to query existing plugins', () => {
                  expect(stepWithPlugins.plugins.filter(p => true).toMatchObject(plugins);
                });
```
thoughts?

Reply:
Good thoughts.